addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1310,3 +1310,13 @@ def execute_command_who(self, **kwargs):
             msg = _("Users in this channel: %(members)s %(dots)s and you.", members=", ".join(members), dots=dots)
 
         self._send_transient_message(self.env.user.partner_id, msg)
+
+    def _prepare_post_data(self, post_data, special_mentions, **kwargs):
+        post_data = super()._prepare_post_data(post_data, special_mentions, **kwargs)
+        if "everyone" in special_mentions:
```

Comment:
This belongs to controller as no-one will ever use it as a model method. This is simply preparing and managing data from a controller so it should be done at controller level (a little of "non modular" approach is acceptable in the controller, better have simple and direct code than bloating models with methods).